### DISCUSSION TITLE
Top-up invoices

### DISCUSSION THREAD

**Discussion by NicolasDorier** [btcpayserver/btcpayserver#2659] [General]
**Motivation**

A common use of BTCPay Server is to top-up the pre-paid account of some service. In such case, asking the user the amount to top up prior to show the invoice seems like a useless step:

The pre-paid account should be top-up with whatever amount of BTC the user paid.

Not only does it removes one step, but:

1. The customer can decrease fees by sending whole UTXO (no change)
2. Without change, it also save on tx fee

**Implementation note**

* When creating an invoice, the amount should be optional. If the amount is not specified, such invoice should be considered top-up.
* When the invoice receives the first payment, it would set its own amount to whatever payment has been received and go to the `processing/paid` status.

**Comment by pavlenex** [comment]
Concept ACK, I like the idea and the use-case it fulfils.

**Comment by NicolasDorier** [comment]
this is now implemented by #2730
